### PR TITLE
Always split image to ABGR image

### LINKED ISSUE #5996: Textures are mangled when building to Windows from macOS
**Describe the bug (REQUIRED)**
Many (all??) of the textures in my game look compressed or mangled when I make a Windows bundle from my Mac. They look fine if I build directly to macOS, or if I copy the project to a Windows machine & build on it directly.

**To Reproduce (REQUIRED)**
1. on a Mac
2. Click on Project -> Bundle -> Windows Application
3. Copy to a Windows machine 

**Expected behavior (REQUIRED)**
I expect to be able to bundle a game on Mac or PC and have it work.

**Defold version (REQUIRED):**
  - Version 1.2.185

**Platforms (REQUIRED):**
 - Platforms: macOS & Windows
 - OS: macOS 11.5 & Windows 10

**Minimal repro case project:**
I've included a sample project with just the star animation shown in the screenshots below (i.e. no background).
[Texture Problem.zip](https://github.com/defold/defold/files/7019559/Texture.Problem.zip)

**Workaround:**
If I build directly on the Windows machine, the problem does not occur.

**Screenshots:**
this is a screenshot from macOS; this is what it *should* look like
macOS screenshot

windows screenshot

### PR DESCRIPTION
**Pull request by britzl**
Fixes #5996

### CODE REVIEW

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/tile/TileSetGenerator.java [hunk 1]
```diff
@@ -161,8 +161,7 @@ private static int calcTileStart(TileSet tileSet, int size, int tileIndex) {
         int tileHeight = tileSet.getTileHeight();
         List<BufferedImage> result = new ArrayList<BufferedImage>(count);
         for (int i = 0; i < count; ++i) {
-            int type = TextureUtil.getImageType(image);
-            BufferedImage tgt = new BufferedImage(tileWidth, tileHeight, type);
+            BufferedImage tgt = new BufferedImage(tileWidth, tileHeight, BufferedImage.TYPE_4BYTE_ABGR);
```
Comment (britzl) [inline com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/tile/TileSetGenerator.java:164]: For indexed (palette) image the problem was this bit of code where we created another indexed image and copied the subimage. The new subimage always seems to result in an image with a black background. I tried setting a transparent background color and clearing the image before copying,  but it didn't help.

Changing to BufferedImage.TYPE_4BYTE_ABGR works and in the end it is the output format for the final texture so we might as well convert it here.

Comment (JCash) [reply com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/tile/TileSetGenerator.java:164]: Is this the only place where we needed to fix it?

> in the end it is the output format for the final texture so we might as well convert it here.

Not sure how to interpret that?
So, even if I input an RGB texture, I'll always get a RGBA texture out?

**Review by JCash — APPROVED** [review]
Approved, with some questions

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/tile/TileSetGenerator.java (hunk 1 above)
Comment (britzl) [reply com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/tile/TileSetGenerator.java:164]: > So, even if I input an RGB texture, I'll always get a RGBA texture out?

Yes, the tilesource image will be split into multiple images, then passed to `TextureSetGenerator.generate()` which will call the `composite()` function. The `composite()` function always creates a `BufferedImage` of type `TYPE_4BYTE_ABGR`:

https://github.com/defold/defold/blob/dev/com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/textureset/TextureSetGenerator.java#L478-L487

Comment (britzl) [reply com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/tile/TileSetGenerator.java:164]: > Is this the only place where we needed to fix it?

Yes, as far as I can tell. For normal atlas creation we do not have this problem (see above).

**Pull request merged**